snuba/optimize.py

```diff
@@ -291,15 +359,20 @@ def optimize_partitions(
 
         query = (query_template % args).strip()
         logger.info(f"Optimizing partition: {partition}")
-        start = time.time()
 
         # Update tracker before running clickhouse.execute since its possible
         # that the connection can get disconnected while the server is still
         # executing the optimization. In that case we don't want to
         # run optimization on the same partition twice.
         if tracker:
-            tracker.update_completed_partitions(partition)
+            tracker.update_scheduled_partitions(partition)
+
+        # if theres a merge in progress for this partition, wait for it to finish
+        if partition in current_merge_info:
+            est_sleep_time = current_merge_info[partition]["estimated_time"]
+            time.sleep(OPTIMIZE_BASE_SLEEP_TIME + est_sleep_time)
```

Comment:
Is there a way to do this without pausing the entire process? As in, move on to another partition while this one finishes?

Reply:
One thing I tried is wrapping this in an async function but then I spoke to @nikhars about that and he said it's actually preferred to pause the entire process because we want to be running only one at a time. He prob can provide extra context here too.